addons/web/controllers/database.py

```diff
@@ -80,7 +80,13 @@ def create(self, master_pwd, name, lang, password, **post):
             country_code = post.get('country_code') or False
             dispatch_rpc('db', 'create_database', [master_pwd, name, bool(post.get('demo')), lang, password, post['login'], country_code, post['phone']])
             credential = {'login': post['login'], 'password': password, 'type': 'password'}
-            request.session.authenticate(name, credential)
+            if request.env:
+                # There is an env of the previous database. Close it as we are creating a new env on the new database
+                # and assign this new env on the request through request.session.authenticate
+                request.env.cr.close()
+            cr = odoo.modules.registry.Registry(name).cursor()
```

Comment:
the cursor isn't closed

Reply:
It is, but it's not clear indeed.

It is assigned on the env which is assigned to the request through `request.session.authenticate`,
and the env/cursor is closed in post_dispatch of the request.

I can maybe add an additional comment as it is really tricky.

Reply:
Ah actually maybe you are right, I will have a second look

Reply:
as a `auth='none'` route there is a chance the request is going through the _serve_nodb` path which doesn't automatically close the cursor, and even if it does the http stack is fine with closed cursors